### PR TITLE
Improve README and docker compose files

### PR DESCRIPTION
**Pull request by samuelim01**
## Description

* Wrote a simple `README.md` for development start up.
* Separated the compose file into base and development versions. This helps prevent issues with empty `node_modules/` folders that may arise when mounted for hot reloading in development environments.

## Checklist

- [x] I have updated documentation
- [x] All tests passing

## Screenshots (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,64 @@
...
+2. Clone the GitHub repository
+```
+git clone https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g03.git
+```
+
+## Getting Started
+
+**Step 1: Copy Environment Configuration File**
+
+To get started, copy the contents of `.env.sample` into a new `.env.` file located at the root level of the project.
```
Comment (limcaaarl) [inline README.md:22]: There's an extra `.` in here `.env.`

**Comment by limcaaarl** [comment]
I tried following the steps in the README, but for some reason I cannot fetch data from question service even though it's up and running.

image

image

**Comment by limcaaarl** [comment]
> I tried following the steps in the README, but for some reason I cannot fetch data from question service even though it's up and running.

Just tested using `docker compose -f compose.yml -f compose.dev.yml up -d` instead of `docker compose -f compose.yml up -d` and it works perfectly fine now.

**Comment by samuelim01** [comment]
> > I tried following the steps in the README, but for some reason I cannot fetch data from question service even though it's up and running.
> 
> Just tested using `docker compose -f compose.yml -f compose.dev.yml up -d` instead of `docker compose -f compose.yml up -d` and it works perfectly fine now.

good catch, there was a very glaring error in the docker file

**Review by limcaaarl — APPROVED** [review]
LGTM!

**Pull request merged**